controller/pkg/extensions/konnect/controlplane.go

```diff
@@ -28,7 +60,126 @@ func ApplyControlPlaneKonnectExtension(ctx context.Context, cl client.Client, co
 		return false, nil
 	}
 
-	// TODO: implement KonnectExtension for ControlPlane v2alpha1: https://github.com/kong/kong-operator/issues/1730
+	config, err := kicInKonnectDefaults(ctx, cl, konnectExtension)
+	if err != nil {
+		return false, fmt.Errorf("failed to generate configuration from KonnectExtension %s/%s for ControlPlane %s/%s: %w",
+			konnectExtension.Namespace, konnectExtension.Name, cp.Namespace, cp.Name, err)
+	}
+	p.KonnectExtensionConfig = config
+
+	// Apply the FillIDs feature gate to the ControlPlane.
+	applyFeatureGatesToControlPlane(cp)
 
 	return true, nil
 }
+
+// GetKonnectConfig returns the KonnectConfig from the KonnectExtensionConfig.
+func (p *ControlPlaneKonnectExtensionProcessor) GetKonnectConfig() *managercfg.KonnectConfig {
+	if p.KonnectExtensionConfig == nil {
+		return nil
+	}
+	return p.KonnectExtensionConfig.KonnectConfig
+}
+
+// kicInKonnectDefaults generates the KonnectExtensionConfig for a KIC instance based on the KonnectExtension resource.
+func kicInKonnectDefaults(ctx context.Context, cl client.Client, konnectExtension *konnectv1alpha2.KonnectExtension) (*KonnectExtensionConfig, error) {
```

Comment:
I know this function is also moved from elsewhere but I think the function name `kicInKonnectConfig()` or `kicInKonnectConfigFromExtension` should describe the usage of the function better. WDYT?